scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -149,252 +196,374 @@ sce_list
-  - Library-specific statistics, e.g. gene-level statistics for a given library found in `rowData`
+1. We should be able to trace library-specific information back to the originating library, including...
   - Cell-level information: Which library is each cell from? 
   Although cells have unique barcode identifiers, these barcodes are only guaranteed to be unique _within_ a library. 
-  Now that we are combining libraries, there's no guarantee we can identify a cell's originating library from its barcode alone.
-- SCE objects should contain the same genes. In other words, each `rowData` table should have the same rownames.
-- SCE columns should match. In other words, the each `colData` should have the same column names.
+  Now that we are merging several libraries, we won't necessarily be able to identify a cell's originating library from its barcode alone.
+  - Library-specific statistics, e.g. gene-level statistics for a given library found in `rowData`
+2. SCE objects should contain the same genes: Each SCE object should have the same row names.
+3. SCE cell metadata columns should match: Each SCE object should have the same column names.
+
+Here is a schematic of what how one of the SCE objects will ultimately be modified into the final merged SCE:
+
+![](diagrams/technical_merge_sce.png)
+
+We'll begin by take some time to thoroughly explore our SCE objects to figure out what wrangling steps we need to take for these specific data.
+Don't skip this exploration!
+Bear in mind that the exact wrangling shown here will not be the same for other SCE objects you work with, but the same general principles apply.
+
+
+#### Preserving library information at the cell level
 
+How will we be able to tell which library a given cell came from?
 
-First, let's determine what the shared genes among SCE objects are using some `purrr` magic.
-Ultimately, we'll filter SCEs down to contain only these genes, thereby ensuring that their genes match.
+First, we'll want to directly add information to the SCE that indicates the different libraries.
+The best way to do this is simply to add a `colData` column with the library information.
+
+In addition, we want to pay some attention to the SCE object's column names (the cell ids), which must remain unique after merging since duplicate ids will cause an R error.
+Note that these ids also correspond to the _row names_ of the `colData` and any reduced dimension matrices.
+Usually (but not always!), the cell ids themselves are cell barcodes, which is the case for our SCE objects (one is shown below as an example):
+
+```{r barcodes}
+# Look at the column names for the `SCPCL000479` library, for example
+colnames(sce_list$SCPCL000479) %>%
+  # Only print out the first 6 for convenience
+  head()
+```
+
+Because barcodes theoretically can be repeated across libraries, we could end up in a situation with non-unique rownames for the `colData`, which would make R complain!
```

Comment:
```suggestion
Because barcodes theoretically can be repeated across libraries, we could end up in a situation with non-unique column names for the SCE object, which would make R complain!
```